File: objectModel/TypeScript/Persistence/CdmFolder/Projections/OperationRenameAttributesPersistence.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

import {
    CdmCorpusContext,
    cdmObjectType,
    CdmOperationRenameAttributes,
    copyOptions,
    cdmLogCode,
    Logger,
    resolveOptions,
} from '../../../internal';
import { OperationRenameAttributes } from '../types';
import { OperationBasePersistence } from './OperationBasePersistence';

/**
 * Operation RenameAttributes persistence
 */
export class OperationRenameAttributesPersistence {
    private static TAG: string = OperationRenameAttributesPersistence.name;

    public static fromData(ctx: CdmCorpusContext, object: OperationRenameAttributes): CdmOperationRenameAttributes {
        if (!object) {
            return undefined;
        }

        const renameAttributesOp: CdmOperationRenameAttributes = OperationBasePersistence.fromData(ctx, cdmObjectType.operationRenameAttributesDef, object);
        renameAttributesOp.renameFormat = object.renameFormat;

        if (typeof (object.applyTo) === 'string') {
            renameAttributesOp.applyTo = [object.applyTo]
        } else if (Array.isArray(object.applyTo)) {
            renameAttributesOp.applyTo = object.applyTo;
        } else if (object.applyTo !== undefined) {
            Logger.error(ctx, this.TAG, this.fromData.name, undefined, cdmLogCode.ErrPersistProjUnsupportedProp);
        }

        return renameAttributesOp;
    }

    public static toData(instance: CdmOperationRenameAttributes, resOpt: resolveOptions, options: copyOptions): OperationRenameAttributes {
        if (!instance) {
            return undefined;
        }

        const data: OperationRenameAttributes = OperationBasePersistence.toData(instance, resOpt, options);
        data.renameFormat = instance.renameFormat;
        data.applyTo = instance.applyTo;

        return data;
    }
}

File: objectModel/Python/cdm/persistence/cdmfolder/projections/operation_rename_attributes_persistence.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License. See License.txt in the project root for license information.

from typing import TYPE_CHECKING

from cdm.enums import CdmObjectType, CdmLogCode
from cdm.persistence.cdmfolder.types import OperationRenameAttributes
from cdm.utilities.logging import logger

from .operation_base_persistence import OperationBasePersistence

if TYPE_CHECKING:
    from cdm.objectmodel import CdmCorpusContext, CdmOperationRenameAttributes
    from cdm.utilities import ResolveOptions, CopyOptions

_TAG = 'OperationRenameAttributesPersistence'


class OperationRenameAttributesPersistence:
    """Operation RenameAttributes persistence"""

    @staticmethod
    def from_data(ctx: 'CdmCorpusContext', data: 'OperationRenameAttributes') -> 'CdmOperationRenameAttributes':
        if not data:
            return None

        rename_attributes_op = OperationBasePersistence.from_data(ctx,
            CdmObjectType.OPERATION_RENAME_ATTRIBUTES_DEF, data)  # type: CdmOperationRenameAttributes
        
        rename_attributes_op.rename_format = data.renameFormat
        if isinstance(data.applyTo, str):
            rename_attributes_op.apply_to = [data.applyTo]
        elif isinstance(data.applyTo, list):
            rename_attributes_op.apply_to = data.applyTo
        elif data.applyTo is not None:
            logger.error(ctx, _TAG, OperationRenameAttributesPersistence.from_data.__name__, None, CdmLogCode.ERR_PERSIST_PROJ_UNSUPPORTED_PROP)

        return rename_attributes_op

    @staticmethod
    def to_data(instance: 'CdmOperationRenameAttributes', res_opt: 'ResolveOptions', options: 'CopyOptions') -> 'OperationRenameAttributes':
        if not instance:
            return None

        obj = OperationBasePersistence.to_data(instance, res_opt, options)  # type: OperationRenameAttributes
        obj.renameFormat = instance.rename_format
        obj.applyTo = instance.apply_to

        return obj
